gadget/mountedfilesystem.go

```diff
@@ -544,12 +578,37 @@ func (f *mountedFilesystemUpdater) Backup() error {
 		return fmt.Errorf("cannot map preserve entries for mount location %q: %v", f.mountPoint, err)
 	}
 
+	knownContent := make(map[string]bool)
+
 	for _, c := range f.ps.ResolvedContent {
+		knownContent[c.Target] = true
 		if err := f.backupVolumeContent(f.mountPoint, &c, preserveInDst, backupRoot); err != nil {
 			return fmt.Errorf("cannot backup content: %v", err)
 		}
 	}
 
+	if f.fromPs != nil {
+		for _, c := range f.fromPs.VolumeStructure.Content {
+			if knownContent[c.Target] {
+				continue
+			}
+
+			destPath, backupPath := f.entryDestPaths(f.mountPoint, "", c.Target, backupRoot)
+			if osutil.IsDirectory(destPath) {
```

Comment:
This would need a comment similar to the one in `mountedFilesystemUpdater`

Reply:
Copied the comment